### PR TITLE
Class implementations

### PR DESCRIPTION
**Pull request by michaelamponsah**
### In this PR, the following requirements have been implemented

- [x] Created  ` Person `,   `Student`, and `Teacher`  classes
- [x] Implemented required methods for the classes above

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  private
+
+  def of_age?()
+    @age >= 18
+  end
+
+  def can_use_services?()
+    of_age? || @parent_permission
+  end
```
Comment (tresorsawasawa) [inline person.rb:19]: - [ ] Kindly ensure the `of_aged?` method is the only private method as right now you define both `can_use_services?` and `of_age?` as `private`. You can just update your code order in the way the `can_use_services?` method come before the `private` word. i.e.
```ruby
  ..
  def can_use_services?()
    of_age? || @parent_permission
  end

  private

  def of_age?()
    @age >= 18
  end
  ...
```

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(classroom)
+    super()
+    @classroom = classroom
+  end
+
```
Comment (tresorsawasawa) [inline student.rb:6]: - [ ] Kindly make sure you correctly implement your code about to Inherit your classes. First, you should import the class file to Inherits from then the current `Constructor` should extend parent's constructor by adding `@classroom` and a `parameter` for it. i.e:
```ruby
require_relative './person.rb'

class Student < Person 
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
  ...
```

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Teacher < Person
+  def initialize(specialization)
+    super()
+    @specialization = specialization
+  end
+
```
Comment (tresorsawasawa) [inline teacher.rb:6]: - [ ] As I mentioned in the above comment, kindly make sure you follow the same approach for the `Teacher` class. You should only change the ``@classroom`` variable to ``@specialization``. i.e.
```ruby
require_relative './person.rb'

class Teacher < Person 
  def initialize(age, specialization, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @specialization = specialization
  end
  ...
```

**Comment by julie-ify** [comment]
Hi @michaelamponsah,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly ensure you fix your linter errors.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by roshan-bajgain — APPROVED** [review]
Hi @michaelamponsah, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Requested change ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**